packages/types/src/chain-adapters/ethereum.ts

```diff
@@ -27,12 +27,20 @@ export type FeeData = {
 }
 
 export type QuoteFeeData = {
-  fee?: string
-  gas?: string
+  /**
+   * estimated gas units in gwei
+   */
   estimatedGas?: string
+  /**
+   * gas price per gwei
+   */
   gasPrice?: string
+  /**
+   * total approval fee in eth
+   */
   approvalFee?: string
-  protocolFee?: string
-  minimumProtocolFee?: string
-  receiveNetworkFee?: string
+  /**
+   * total fee including approval
+   */
+  totalFee?: string
```

Comment:
formally, storing totalFee here violates 2NF, which means that it's possible to construct an inconsistent and meaningless QuoteFeeData. practically, it means that `QuoteFeeData` must never be used as the _input_ to a function without an extra (runtime) validity check that will probably be overlooked. TS doesn't really have a great way to forbid a type being used as a parameter, and doing so would break composability and probably be a bad idea for that reason alone anyhow.

A better option would be to export a function which takes a QuoteFeeData and computes the totalFee. If that's not ergonomic enough, the logic could be baked into a getter and QuoteFeeData could become a specific full class instead of just the interface type, though that approach has its own drawbacks.